cilium/cmd/kernel_check.go

```diff
@@ -0,0 +1,42 @@
+// Copyright 2020 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"github.com/cilium/cilium/pkg/datapath/linux/probes"
+	"github.com/spf13/cobra"
+)
+
+var kernelCheckCmd = &cobra.Command{
+	Use:   "kernel-check",
+	Short: "Checks whether machine is supporting required kernel capabilities",
+	Run: func(cmd *cobra.Command, args []string) {
+		kernelCheck()
+	},
+}
+
+func kernelCheck() {
+	probeManager := probes.NewProbeManager()
+	if err := probeManager.SystemConfigProbes(); err != nil {
+		log.WithError(err).Warning("BPF system config check: NOT OK")
+	}
+	if err := probeManager.CreateHeadersFile(); err != nil {
+		log.WithError(err).Fatal("BPF check: NOT OK.")
+	}
```

Comment:
When I looked at the code again, I realized this bit doesn't really make sense in the context of this CLI. It can be removed.